.github/workflows/ci-full.yml

```diff
@@ -0,0 +1,767 @@
+name: CI Full suite
+# This is a full workflow for the production branches and tags, which
+# includes building and testing on all supported platforms, and
+# uploading the artifacts to S3.
```

Comment:
One thing I'd like to add (later sometime) is being able to trigger all builds for a branch, like here: https://github.com/redis/redis/blob/7c9f41b52b3a664c07581c83f9256fbedb798dfe/.github/workflows/daily.yml#L10

So, does it make sense to separate regular test builds and publish/upload runs? Maybe we create ci-publish.yml and have builds that publish/upload there?